recipe/fix_cub_constexpr.diff

```diff
@@ -0,0 +1,18 @@
+diff --git a/cupy/cuda/cupy_cub.cu b/cupy/cuda/cupy_cub.cu
+index 802d62bd5..3b230e8d1 100644
+--- a/cupy/cuda/cupy_cub.cu
++++ b/cupy/cuda/cupy_cub.cu
+@@ -85,8 +85,11 @@ class numeric_limits<__half> {
+   public:
+     static __host__ __device__ constexpr __half infinity() noexcept {
+         unsigned short inf_half = 0x7C00U;
+-        __half inf_value = *reinterpret_cast<__half*>(&inf_half);
+-        return inf_value;
++        // WAR:
++        // - we want a constexpr here, but reinterpret_cast cannot be used
++        // - we want to use std::bit_cast, but it requires C++20 which is too new
++        // - we use the compiler builtin, fortunately both gcc and msvc have it
++        return __builtin_bit_cast(__half, inf_half);
```

Comment:
Thanks to @miscco  for suggesting the WAR. 
https://github.com/microsoft/STL/blob/0515a05b394596de92d08cb0f352614479a2a883/stl/inc/type_traits#L2477-L2478

Reply:
https://godbolt.org/z/qzPqd5z6E

Reply:
btw @kmaehashi was CuPy on Windows built by vs2015 not 2017? I am wondering why this was not caught in the CI...